### PR TITLE
Adds hot reloading/zero downtime deploy

### PR DESCRIPTION
**Pull request by tomdale**
This commit adds support for reloading a running FastBoot server.

In the programmatic case, you can call `server.reload()` to have it
destroy the old app and re-read the new version off the disk.

If running from the command line, you can send SIGUSR1 to the process
and it will reload the app from disk.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -40,7 +40,10 @@
   "devDependencies": {
     "chai": "^3.5.0",
     "chai-as-promised": "^5.2.0",
+    "fs-extra": "^0.26.7",
```
Comment (rwjblue) [inline package.json]: This isn't used in this PR

Comment (tomdale) [inline package.json]: It's used here: https://github.com/ember-fastboot/ember-fastboot-server/pull/39/files#diff-4b704b373a591b75fa93036c2f681941R95

Comment (rwjblue) [inline package.json]: @tomdale - That is using `fs-promise`, not `fs-extra`.

Comment (tomdale) [inline package.json]: @rwjblue You appear to be correct.

**Pull request merged**